### PR TITLE
MES-950: Added git dependency

### PR DESCRIPTION
**Pull request by benitezfede**
Added the git dependency to fix the issue collecting looker views.

Tested locally generating the agent lambda image:

```
$ docker build -t lambda_agent --target lambda --platform=linux/amd64 . 
```

Run the image locally:
```
$ docker run --rm --name lambda_agent --entrypoint /bin/sh -it lambda_agent  
```

Test git clone with script:
```
$ cat <<EOF > ./test_git_clone.py
import git
import os
import tempfile
import base64

def clone_repository(repo_url, ssh_key_string):
    try:
        # Create a temporary file to store the SSH key
        with tempfile.NamedTemporaryFile(delete=False) as ssh_key_file:
            ssh_key = base64.b64decode(ssh_key_string)

            print(f"Writing SSH key: {ssh_key}")

            ssh_key_file.write(ssh_key)
            ssh_key_path = ssh_key_file.name

        # Set up SSH key environment for Git
        os.environ['GIT_SSH_COMMAND'] = f"ssh -i {ssh_key_path} -o StrictHostKeyChecking=no"

        # Define the directory to clone the repository to
        clone_dir = "/tmp/repo"

        ssh_options = " ".join(
            [
                "-v",  # Verbose helps in case of problems
                "-o StrictHostKeyChecking=no",  # We do not know all the possible hosts, so do not check
                "-o UserKnownHostsFile=/dev/null",  # Do not write known_hosts
                "-o GlobalKnownHostsFile=/dev/null",  # Do not write known_hosts
                f"-i {ssh_key_path}",  # Use this key
            ]
        )

        repo_url = repo_url.lstrip("https://")

        env = {**os.environ, **{"GIT_SSH_COMMAND": f"ssh {ssh_options}"}}

        # Clone the repository
        print(f"Cloning repository from {repo_url} to {clone_dir}...")
        # Use depth 1 to bring only the latest revision.
        git_params = ["clone", "--depth", "1", repo_url, clone_dir]
        git.exec_command(*git_params, env=env)

        print("Repository cloned successfully.")
    except Exception as e:
        print(f"Failed to clone repository: {e}")


if __name__ == "__main__":
    # Replace with your repository URL and SSH key as a string
    REPO_URL = "git@github.com:monte-carlo-data/looker.git"
    SSH_KEY_STRING = "XXXXX"

    clone_repository(REPO_URL, SSH_KEY_STRING)
EOF

$ python test_git_clone.py 

Cloning repository from git@github.com:monte-carlo-data/looker.git to /tmp/repo...
INFO:git:executing git command: "git clone --depth 1 git@github.com:monte-carlo-data/looker.git /tmp/repo"
Repository cloned successfully.
```

### CODE REVIEW

**Review by lexicalunit — APPROVED** [review]
Rad! 🤘

File: Dockerfile [hunk 1]
```diff
@@ -101,8 +94,10 @@ RUN pip install --no-cache-dir --upgrade pip
 COPY --from=lambda-builder "${LAMBDA_TASK_ROOT}" "${LAMBDA_TASK_ROOT}"
 
 # install unixodbc and 'ODBC Driver 17 for SQL Server', needed for Azure Dedicated SQL Pools
+# install git needed for looker views collection
```
Comment (mdediana) [inline Dockerfile:97]: Nit: It's used to collect more than looker views: https://github.com/monte-carlo-data/common/blob/9f1b1eb3d4caf19fa21524504597575d97b14562/montecarlodata_common/object_types.py#L8-L12

**Review by mrostan — APPROVED** [review]
looks good to me, thanks!

**Review by mdediana — APPROVED** [review]
LGTM

File: Dockerfile (hunk 1 above)
Comment (benitezfede) [reply Dockerfile:97]: The other types are retrieved using the API [here](https://github.com/monte-carlo-data/data-collector/blob/8931a42a4aee0e01a2b1a8c501caea5a15ca51e4/lambdas/clients/looker/looker_proxy_client.py#L112)

**Pull request merged**